specifications/dutch_auction.md

```diff
@@ -0,0 +1,261 @@
+[English auction](https://en.wikipedia.org/wiki/English_auction)
+(an open-outcry ascending price auction)
+comes from the advantage that the Dutch auction is over
+after the first bid;
+thus, it is not necessary to store the bids of potential validators.
+
+### Goals
+ *  Establish a mechanism to determine the price to become a validator.
+
+### Non-Goals
+ *  The mechanism to become a validator.
+ *  The specifics of the ETHDKG protocol.
+
+### Assumptions
+Values defined at deployment are a guide
+and may be modified in the future.
+
+## Specification
+
+### Overview
+The solution consists of a Smart Contract that provides the following functionality:
+
+#### Start an auction
+This operation starts an auction by calculating the initial and final prices
+and registering the current block number as the auction's start block;
+upon execution, an event with all auction information will be emitted.
+The execution of this operation will be triggered
+only by the sidechain (onlyFactory)
+whenever a new validator position is opened. 
+
+#### Get current bidding price
+This operation calculates the current bidding price
+as determined by the price curve and the number of blocks
+since the start of the auction.
+The execution of this operation may be performed at any time
+to check the current bid price.
+
+#### Bid for current auction price
+This operation emits an event with address of the winner
+and auction details.
+The execution of this operation can be triggered at any time
+by any address that pays the current bid price to become a validator. 
+
+### Data
+These variables govern the price curve,
+and most of them are defined before contract deployment;
+only the auction's final price is defined at execution time
+since depends on the gas price of the network
+and the number of current validators at auction start. 
+
+#### Deployment variables
+The following values are set at deployment to produce the results detailed
+in the [Testing](#testing) section;
+these values may be modified upon contract redeployment to adapt the price curve
+to specific requirements.
+
+| Descriptor | Description | Initial Constructor Value |
+| ----------- | ----------- |----------- |
+| Start Price  | The initial price for auction in Wei | 1000000 * 10 ** 18 |
+| Decay | The decay factor (how fast bidding price decreases with time) | 16 |
+| Scale Parameter | The scale factor (how curve is compressed) | 100 |
+
+#### Execution variables
+The following values are calculated at execution time.
+
+| Descriptor | Description | Calculated Value |
+| ----------- | ----------- |----------- |
+| Final Price  | The final price for auction in Wei (bidding price can never be less than this value) | Cost to add a validator * Current Number of Registered Validators|
+
+The cost to add a validator is calculated as follows:
+ *  ETHDKG Single Validator Cost (Two ETHDKG units operations 1.2M gas units)
+    -> 1200000 * 2 operations (In and Out) * gasPriceInWei
```

Comment:
What do you mean by "In and Out"?

Reply:
That would be: destroying the current key (out) and generating the new one (in).